oak_runtime/src/introspect.rs

```diff
@@ -111,20 +111,47 @@ fn find_client_file(path: &str) -> Option<(Vec<u8>, String)> {
         .as_str();
 
     match subpath {
-        "/index.js" => Some((
-            include_bytes!("../introspection_browser_client/dist/index.js.gz").to_vec(),
-            "application/javascript".to_string(),
-        )),
-        "/graphvizlib.wasm" => Some((
-            include_bytes!("../introspection_browser_client/dist/graphvizlib.wasm.gz").to_vec(),
-            "application/wasm".to_string(),
-        )),
+        "/index.js" => {
+            let mut response = Response::new(Body::from(
+                include_bytes!("../introspection_browser_client/dist/index.js.gz").to_vec(),
+            ));
+            let headers = response.headers_mut();
+            headers.insert(CONTENT_TYPE, "application/javascript".parse().unwrap());
+            headers.insert(hyper::header::CONTENT_ENCODING, "gzip".parse().unwrap());
+
+            Some(response)
+        }
+        "/graphvizlib.wasm" => {
+            let mut response = Response::new(Body::from(
+                include_bytes!("../introspection_browser_client/dist/graphvizlib.wasm.gz").to_vec(),
+            ));
+            let headers = response.headers_mut();
+            headers.insert(CONTENT_TYPE, "application/wasm".parse().unwrap());
+            headers.insert(hyper::header::CONTENT_ENCODING, "gzip".parse().unwrap());
+
+            Some(response)
+        }
+        "/favicon.png" => {
+            let mut response = Response::new(Body::from(
+                include_bytes!("../introspection_browser_client/dist/favicon.png").to_vec(),
+            ));
```

Comment:
May be useful to define some helper method(s) so that you can just do something like `Some(static_file("favicon.png", "application/png"))` etc.

Reply:
Good point, did that :). I'm always reluctant to create abstractions, but it does make things easier here.